src/Microsoft.PowerShell.IoT/Microsoft.PowerShell.IoT.cs

```
@@ -316,26 +366,36 @@ public SendSPIData()
-            if (this.Channel == 1)
-            {
-                spiChannel = Unosquare.RaspberryIO.Pi.Spi.Channel1;
-                Unosquare.RaspberryIO.Pi.Spi.Channel1Frequency = (int)this.Frequency;
-            }
-            else
+            try
             {
-                Unosquare.RaspberryIO.Pi.Spi.Channel0Frequency = (int)this.Frequency;
-            };
+                var spiChannel = Unosquare.RaspberryIO.Pi.Spi.Channel0;
+                if (this.Channel == 1)
+                {
+                    spiChannel = Unosquare.RaspberryIO.Pi.Spi.Channel1;
+                    Unosquare.RaspberryIO.Pi.Spi.Channel1Frequency = (int)this.Frequency;
+                }
+                else
+                {
+                    Unosquare.RaspberryIO.Pi.Spi.Channel0Frequency = (int)this.Frequency;
+                };
 
-            var response = spiChannel.SendReceive(this.Data);
-            if (this.Raw)
-            {
-                WriteObject(response);
+                var response = spiChannel.SendReceive(this.Data);
+                if (this.Raw)
+                {
+                    WriteObject(response);
+                }
+                else
+                {
+                    SPIData spiData = new SPIData(this.Channel, this.Frequency, this.Data, response);
+                    WriteObject(spiData);
+                }
             }
-            else
+            catch (System.TypeInitializationException e) // Unosquare.RaspberryIO.Gpio.GpioController.Initialize throws this TypeInitializationException
             {
-                SPIData spiData = new SPIData(this.Channel, this.Frequency, this.Data, response);
-                WriteObject(spiData);
+                if (!Unosquare.RaspberryIO.Computer.SystemInfo.Instance.IsRunningAsRoot)
+                {
+                    throw new PlatformNotSupportedException(Resources.ErrNeedRootPrivileges, e);
+                }
             }
```

If they are running as root we should probably `throw` the error again so we don't accidentally swallow it?